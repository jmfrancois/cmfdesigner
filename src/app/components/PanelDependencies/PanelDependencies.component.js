import React from 'react';
import PropTypes from 'prop-types';
import { Inject } from '@talend/react-cmf';

class PanelDependencies extends React.Component {
	constructor(props) {
		super(props);
		this.state = { opened: false };
		this.onClick = this.onClick.bind(this);
	}

	onClick(event) {
		event.preventDefault();
		this.setState({ opened: !this.state.opened });
	}

	render() {
		if (this.props.dependencies.length === 0) {
			return null;
		}
		return (
			<div>
				<h3>
					<a href="#open-imports" onClick={this.onClick}>
						<Inject component="Icon" componentId={this.state.opened ? 'selection-list-opened' : 'selection-list-closed'} />
						Imports
					</a>
				</h3>
				{this.state.opened && (
					<ul>
						{this.props.dependencies.map((dep, index) => (
							<li key={index}>
								import {!dep.default && '{'} {dep.name} {!dep.default && '}'} from '{dep.source}';
							</li>
						))}
					</ul>
				)}
			</div>

		);
	}
}

PanelDependencies.propTypes = {
	dependencies: PropTypes.arrayOf(PropTypes.shape({
		default: PropTypes.bool,
		name: PropTypes.string,
		source: PropTypes.string,
	})),
};
PanelDependencies.defaultProps = {
	dependencies: [],
};
PanelDependencies.displayName = 'PanelDependencies';

export default PanelDependencies;
