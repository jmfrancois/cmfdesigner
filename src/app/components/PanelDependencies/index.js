import PanelDependencies from './PanelDependencies.component';

export default PanelDependencies;
