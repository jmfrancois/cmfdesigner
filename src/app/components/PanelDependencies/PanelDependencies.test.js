import React from 'react';
import { shallow } from 'enzyme';

import PanelDependencies from './PanelDependencies.component';

describe('PanelDependencies', () => {
	it('should render', () => {
		const wrapper = shallow(
			<PanelDependencies />
		);
		expect(wrapper.getElement()).toMatchSnapshot();
	});
});
