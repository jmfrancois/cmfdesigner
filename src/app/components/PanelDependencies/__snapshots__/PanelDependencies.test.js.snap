// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PanelDependencies should render 1`] = `null`;
